apiserver/facades/client/application/deployrepository.go

```diff
@@ -569,7 +569,7 @@ func (v *deployFromRepositoryValidator) resolveCharm(curl *charm.URL, requestedO
 	}
 
 	// Get the series to use.
-	series, err := selector.CharmSeries()
+	series, _, err := selector.CharmSeries()
```

Comment:
The same check is required for the current Deploy facade as well since it can be called without the juju client by terraform or via the python library.

Reply:
I changed the implementation a little bit, now it's done in `Validate()`